packages/api-server/src/base/gw-config.ts

```diff
@@ -0,0 +1,351 @@
+import { core, utils, HexNumber, Script, HexString } from "@ckb-lumos/base";
+import { normalizers } from "@ckb-lumos/toolkit";
+import {
+  BackendInfo,
+  EoaScript,
+  GwScript,
+  NodeInfo,
+  RollupCell,
+  RollupConfig,
+} from "./types/node-info";
+import {
+  NodeMode,
+  BackendType,
+  EoaScriptType,
+  GwScriptType,
+  GodwokenClient,
+  NodeInfo as GwNodeInfo,
+} from "@godwoken-web3/godwoken";
+import { CKB_SUDT_ID } from "../methods/constant";
+import { Uint32 } from "./types/uint";
+import { snakeToCamel } from "../util";
+
+export class GwConfig {
+  rpc: GodwokenClient;
+  private nodeInfo: NodeInfo | undefined;
+
+  web3ChainId: HexNumber | undefined;
+  accounts: ConfigAccounts | undefined;
+  eoaScripts: ConfigEoaScripts | undefined;
+  backends: ConfigBackends | undefined;
+  gwScripts: ConfigGwScripts | undefined;
+  rollupConfig: RollupConfig | undefined;
+  rollupCell: RollupCell | undefined;
+  nodeMode: NodeMode | undefined;
+  nodeVersion: string | undefined;
+
+  constructor(rpcOrUrl: GodwokenClient | string) {
+    if (typeof rpcOrUrl === "string") {
+      this.rpc = new GodwokenClient(rpcOrUrl);
+      return;
+    }
+
+    this.rpc = rpcOrUrl;
+  }
+
+  async getNodeInfo() {
+    const nodeInfo = await this.rpc.getNodeInfo();
+    return toApiNodeInfo(nodeInfo);
+  }
+
+  init(
+    successCallback?: (gwConfig: GwConfig) => any,
+    errCallBack?: (error: any) => any
+  ): Promise<GwConfig> {
+    return new Promise(async (resolve, reject) => {
+      successCallback ||= (_gwConfig: GwConfig) => {};
+      errCallBack ||= (_error: any) => {};
+      try {
+        this.nodeInfo ||= await this.getNodeInfo();
```

Comment:
Another why to set `this.nodeInfo` ?

If not, we can replace `||=` with `=` in following fields.

```suggestion
        this.nodeInfo = await this.getNodeInfo();
```